generated/nidigital/session.py

```diff
@@ -2676,6 +2676,88 @@ def export_signal(self, signal, signal_identifier, output_terminal):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return
 
+    @ivi_synchronized
+    def _fetch_capture_waveform(self, site_list, waveform_name, samples_to_read, timeout):
+        # This is slightly modified codegen from the function
+        # We cannot use codegen without major modifications to the code generator
+        # This function uses two 'ivi-dance' parameters and then multiplies them together - see
+        # the (modified) line below
+        # Also, we want to return the two sized that normally wouldn't be returned
+        vi_ctype = _visatype.ViSession(self._vi)  # case S110
+        site_list_ctype = ctypes.create_string_buffer(site_list.encode(self._encoding))  # case C020
+        waveform_name_ctype = ctypes.create_string_buffer(waveform_name.encode(self._encoding))  # case C020
+        samples_to_read_ctype = _visatype.ViInt32(samples_to_read)  # case S150
+        timeout_ctype = _converters.convert_timedelta_to_seconds(timeout, _visatype.ViReal64)  # case S140
+        data_buffer_size_ctype = _visatype.ViInt32(0)  # case S190
+        data_ctype = None  # case B610
+        actual_num_waveforms_ctype = _visatype.ViInt32()  # case S220
+        actual_samples_per_waveform_ctype = _visatype.ViInt32()  # case S220
+        error_code = self._library.niDigital_FetchCaptureWaveformU32(vi_ctype, site_list_ctype, waveform_name_ctype, samples_to_read_ctype, timeout_ctype, data_buffer_size_ctype, data_ctype, None if actual_num_waveforms_ctype is None else (ctypes.pointer(actual_num_waveforms_ctype)), None if actual_samples_per_waveform_ctype is None else (ctypes.pointer(actual_samples_per_waveform_ctype)))
+        errors.handle_error(self, error_code, ignore_warnings=True, is_error_handling=False)
+        data_buffer_size_ctype = _visatype.ViInt32(actual_num_waveforms_ctype.value)  # case S200
```

Comment:
This should be `actual_num_waveforms_ctype.value * actual_samples_per_waveform_ctype.value`

Reply:
Fixed